packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java

```diff
@@ -400,6 +429,39 @@ public void createRuleWithActions_adminCanUsePrivateAction()
         restClient.assertStatusCodeIs(CREATED);
     }
 
+    /**
+     * Check that an administrator can create rules with email (private) action with reference to an email template.
+     */
+    @Test
+    public void createRuleWithActions_adminCanUseMailActionWithTemplate()
+    {
+        final RestRuleModel ruleModel = rulesUtils.createRuleModelWithDefaultValues();
+        final RestActionBodyExecTemplateModel mailAction = new RestActionBodyExecTemplateModel();
+        mailAction.setActionDefinitionId(MAIL_ACTION);
+        final Map<String, Serializable> params = new HashMap<>();
+        final UserModel sender = getRandomUserModel();
+        final UserModel recipient = getRandomUserModel();
+        params.put("from", sender.getEmailAddress());
+        params.put("to", recipient.getEmailAddress());
+        params.put("subject", "Test");
+        final RestActionDefinitionModel actionDef =
+                restClient.authenticateUser(user).withCoreAPI().usingActions().getActionDefinitionById(MAIL_ACTION);
+        final RestParameterDefinitionModel paramDef =
+                actionDef.getParameterDefinitions().stream().filter(param -> param.getName().equals(TEMPLATE_PARAM)).findFirst().get();
+        final String constraintName = paramDef.getParameterConstraintName();
+        final RestActionConstraintModel constraint =
+                restClient.authenticateUser(user).withCoreAPI().usingActions().getActionConstraintByName(constraintName);
+        String templateScriptRef = constraint.getConstraintValues().stream().findFirst().get().getValue();
```

Comment:
I've added some code in the rule util for this - getting a constraint value.  It's on a branch later than this one though, so we just need to remember to tidy it up later on.

Reply:
Actually it's being used lower down. See `findConstraintValue` below.

Reply:
Thank, I may try to include that in this PR as I already have some fixes to do.

Reply:
BTW, I merged the "later branch" you mention into this one, so the util code is there, however I removed several tests you added as I believe they weren't up to date with our latest decisions on validations.

Reply:
Ah great - thanks!